src/Aspire.Hosting.Qdrant/QdrantBuilderExtensions.cs

```diff
@@ -0,0 +1,98 @@
+            .WithEnvironment(context =>
+            {
+                context.EnvironmentVariables[ApiKeyEnvVarName] = qdrant.ApiKeyParameter;
+
+                // If in Publish mode, disable static content, which disables the Dashboard Web UI
+                // https://github.com/qdrant/qdrant/blob/acb04d5f0d22b46a756b31c0fc507336a0451c15/src/settings.rs#L36-L40
+                if (builder.ExecutionContext.IsPublishMode)
+                {
+                    context.EnvironmentVariables[DisableStaticContentEnvVarName] = "0";
+                }
+            });
+    }
+
+    /// <summary>
+    /// Adds a named volume for the data folder to a Qdrant container resource.
+    /// </summary>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="name">The name of the volume. Defaults to an auto-generated name based on the resource name.</param>
+    /// <param name="isReadOnly">A flag that indicates if this is a read-only volume.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<QdrantServerResource> WithDataVolume(this IResourceBuilder<QdrantServerResource> builder, string? name = null, bool isReadOnly = false)
+        => builder.WithVolume(name ?? VolumeNameGenerator.CreateVolumeName(builder, "data"), "/qdrant/storage", isReadOnly);
+
+    /// <summary>
+    /// Adds a bind mount for the data folder to a Qdrant container resource.
+    /// </summary>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="source">The source directory on the host to mount into the container.</param>
+    /// <param name="isReadOnly">A flag that indicates if this is a read-only mount.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<QdrantServerResource> WithDataBindMount(this IResourceBuilder<QdrantServerResource> builder, string source, bool isReadOnly = false)
+        => builder.WithBindMount(source, "/qdrant/storage", isReadOnly);
+
+    /// <summary>
+    /// Add a reference to a Qdrant settings for a project.
+    /// </summary>
+    /// <param name="builder">An <see cref="IResourceBuilder{T}"/> for <see cref="ProjectResource"/></param>
+    /// <param name="qdrantResource">The Qdrant server resource</param>
+    /// <returns></returns>
+    public static IResourceBuilder<ProjectResource> WithReference(this IResourceBuilder<ProjectResource> builder, IResourceBuilder<QdrantServerResource> qdrantResource)
+    {
+        builder.WithEnvironment(context =>
+        {
+            context.EnvironmentVariables[$"ConnectionStrings__{qdrantResource.Resource.Name}"] = new ConnectionStringReference(qdrantResource.Resource, optional: false);
+
+            var restEndpointUrl = $"Endpoint={qdrantResource.Resource.GetEndpoint("rest").Url};Key={qdrantResource.Resource.ApiKeyParameter.Value}";
+            if (restEndpointUrl is not null)
```

Comment:
This will never be null?

Reply:
yeah we need a new API for this.

Reply:
I realize I'm checking the wrong thing for null, but is it really that `.Exists()` is just going to be a null check anyway?

```
var restEndpoint = qdrantResource.Resource.GetEndpoint("rest");
if (restEndpoint is not null) //Exists?
{
    context.EnvironmentVariables[$"ConnectionStrings__{qdrantResource.Resource.Name}_rest"] = $"Endpoint={restEndpoint.Url};Key={qdrantResource.Resource.ApiKeyParameter.Value}";
}
```

or more a `TryGetEndpoint`?

Reply:
it would make sure there's a backing endpoint annotation. Since this change is in main, maybe we just expose it in this PR.

Reply:
It's possible for an `EndpointReference` on a `Resource` to not have annotations?!

Reply:
The EndpointReference is a promise so it never returns null. You get the reference and asks questions about it 😄 .

Reply:
Wait, so is it already there? `IsAllocated` seems to validate the annotation exists...

Reply:
Right that's if the endpoint is allocated, I guess you can use that for this scenario. Exists is different. Exists says do I have the EndpointAnnotation. IsAllocated tells you if the endpoint is allocated by DCP. In publish mode, none of the endpoints are allocated, but they all exist.

Reply:
Wait is this endpoint ever not defined?

Reply:
No it always is defined.  The resource is created always with 2 endpoints, primary and this named one.
```
return builder.AddResource(qdrant)
    .WithImage(QdrantContainerImageTags.Image, QdrantContainerImageTags.Tag)
    .WithHttpEndpoint(hostPort: port, containerPort: QdrantPortGrpc, name: QdrantServerResource.PrimaryEndpointName)
    .WithHttpEndpoint(hostPort: port, containerPort: QdrantPortHttp, name: "rest")

// other parts snipped
```

Maybe that makes you feel better about the simpler check?

But if not, is this what you had in mind for the Exists? (virtually same as Allocated, just not checking the Allocated check?

```
public bool Exists => GetAnnotation() is not null;

private EndpointAnnotation GetAnnotation()
{
    var endpoint = _endpointAnnotation ??= Resource.Annotations.OfType<EndpointAnnotation>().SingleOrDefault(a => StringComparers.EndpointAnnotationName.Equals(a.Name, EndpointName));
    ArgumentNullException.ThrowIfNull(endpoint, $"The endpoint `{EndpointName}` does not exist for the resource `{Resource.Name}`.");
    return endpoint;
}
```